### PR TITLE
Add team feature fake door

### LINKED ISSUE #1554: Fake Door for Teams
Screen Shot 2021-09-21 at 1 42 29 PM

This view is meant to validate interest in our team features. To separate those who are truly interested from those clicking a novel button, we added a button "Sign me up!" which takes us to a 1 question blocksurvey form to optionally leave contact information. 

I suppose we'll want to track the click through rate of the button and compare that to our DAU.

### PR DESCRIPTION
**Pull request by Tbaut**
closes #1554

I couldn't test Posthog integration because it's down for the moment.

### CODE REVIEW

**Review by FSM1 — APPROVED** [review]

File: packages/files-ui/src/Components/Layouts/AppHeader.tsx [hunk 1]
```diff
@@ -186,6 +192,11 @@ const AppHeader = ({ navOpen, setNavOpen }: IAppHeader) => {
     window.open(ROUTE_LINKS.DiscordInvite, "_blank")
   }, [posthog])
 
+  const onStartATeamClick = useCallback(() => {
+    posthog && posthog.capture("Team start button click")
```
Comment (FSM1) [inline packages/files-ui/src/Components/Layouts/AppHeader.tsx]:
```suggestion
    posthog?.capture("Team start button click")
```

File: packages/files-ui/src/Components/Elements/TeamModal.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+interface Props {
+    onHide: () => void
+}
+
+const TeamModal = ({ onHide }: Props) => {
+  const classes = useStyles()
+  const posthog = usePosthog()
+
+  const onSignupTeamClick = useCallback(() => {
+    posthog && posthog.capture("Team signup button click")
```
Comment (FSM1) [inline packages/files-ui/src/Components/Elements/TeamModal.tsx]:
```suggestion
    posthog?.capture("Team signup button click")
```

**Review by FSM1 — APPROVED** [review]
just some minor nitpicks

File: packages/files-ui/src/Components/Layouts/AppHeader.tsx [hunk 3]
```diff
@@ -149,7 +150,11 @@ const useStyles = makeStyles(
         margin: `0 ${constants.generalUnit * 2}px`,
 
         "& button" : {
-          height: constants.generalUnit * 4
+          height: constants.generalUnit * 4,
+
+          "&:not(:first-child)": {
```
Comment (RyRy79261) [inline packages/files-ui/src/Components/Layouts/AppHeader.tsx:154]: I love this

**Review by RyRy79261 — APPROVED** [review]

File: packages/files-ui/src/Components/Layouts/AppHeader.tsx (hunk 1 above)
Comment (Tbaut) [reply packages/files-ui/src/Components/Layouts/AppHeader.tsx]: I'm actually going to remove this all, found a way to track things without explicit posthog calls.

Comment (FSM1) [reply packages/files-ui/src/Components/Layouts/AppHeader.tsx]: yeah PH is pretty good at figuring out which elements were clicked. should not really require much manual work besides specifying the element in the PH dashboard

**Review by FSM1 — APPROVED** [review]

**Pull request merged**